www/js/lib/zimArchive.js

```diff
@@ -71,153 +71,117 @@ var LZ;
-                    that.callLibzimWorker({ action: 'init', files: that.file._files }).then(function () {
-                        that.libzimReady = 'ready';
-                        // If user is using libzim for reading the file, we have delayed the callback till now
-                        if (params.useLibzim) whenZimReady();
-                        params.searchProvider = 'fulltext: ' + libzimReaderType;
-                        // Update the API panel
-                        uiUtil.reportSearchProviderToAPIStatusPanel(params.searchProvider);
-                    }).catch(function (err) {
-                        uiUtil.reportSearchProviderToAPIStatusPanel(params.searchProvider + ': ERROR');
-                        console.error('The libzim worker could not be instantiated!', err);
-                        that.libzimReady = 'error';
-                    });
-                } else {
-                    // var message = 'Full text searching is not available because ';
-                    if (!that.file.fullTextIndex) {
-                        params.searchProvider += ': no_fulltext'; // message += 'this ZIM does not have a full-text index.';
-                    } else if (isSplitZim) {
-                        params.searchProvider += ': split_zim'; // message += 'the ZIM archive is split.';
-                    } else if (typeof Atomics === 'undefined') {
-                        params.searchProvider += ': no_atomics'; // message += 'this browser does not support Atomic operations.';
-                    } else if (/Android/.test(params.appType)) {
-                        params.searchProvider += ': no_sharedArrayBuffer';
-                    } else if (params.debugLibzimASM === 'disable') {
-                        params.searchProvider += ': disabled';
-                    } else {
-                        params.searchProvider += ': unknown';
-                    }
-                    uiUtil.reportSearchProviderToAPIStatusPanel(params.searchProvider);
-                }
-                // Set the archive file type ('open', 'zimit' or 'zimit2')
-                return that.addMetadataToZIMFile('Scraper').then(function () {
-                    params.zimType = that.setZimType();
-                    // If user is not using libzim for reading the file, we can call the ready callback now
-                    if (!params.useLibzim) whenZimReady();
-                });
-            }).catch(function (err) {
-                console.warn('Error setting archive listings: ', err);
-            });
-        });
-    };
-    if (storage && !path) {
-        var fileList = storage;
-        // We need to convert the FileList into an Array
-        var fileArray = [].slice.call(fileList);
-        // The constructor has been called with an array of File/Blob parameter
-        createZimfile(fileArray);
+function ZIMArchive(storage, path, callbackReady, callbackError) {
```

Comment:
What concrete code changes have you made to the ZIMArchive object definition?  There are a mass of changes that are not explained, and it's impossible for me to see concretely what has changed in substance.